### PR TITLE
Feat/add block detail card test

### PR DESCRIPTION
**Pull request by Thistle-Strum**
🔥 Summary
Adds tests for block detail card component https://github.com/casper-network/casper-blockexplorer-frontend/issues/316

### CODE REVIEW

File: app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+  return {
+    useTranslation: () => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-unsafe-member-access
+      return { t: (key: string) => englishTranslations[key] };
+    },
+  };
+});
+
+const getMockBlock = () => ({
```
Comment (mgibson-casper) [inline app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx]: Could we please move this to another file? It's probably best to have a `/mocks` folder or something

File: app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    expect(baseCard).toBeInTheDocument();
+    expect(baseCard).toContainElement(baseCardBody);
+  });
+
+  it('should render the BaseCard body content', () => {
+    render(<BlockDetailsCard block={getMockBlock()} isLoading={false} />);
+    const { hash, header, body } = getMockBlock();
+
+    const baseCardBody = screen.getByTestId('baseCardBody');
+    const blockHeight = screen.getByTestId('block-height');
```
Comment (mgibson-casper) [inline app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx:50]: Are any of these links? If so, we should test them as such.

File: app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx (hunk 1 above)
Comment (DorianKwan) [reply app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx]: +1

File: app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx (hunk 2 above)
Comment (Thistle-Strum) [reply app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx:50]: Added `getByRole('link', {name: ...})` for links

File: app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx (hunk 1 above)
Comment (Thistle-Strum) [reply app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx]: Moved `getMockBlock()` but still working on a solution for the jest mocks

File: app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx (hunk 2 above)
Comment (mgibson-casper) [reply app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx:50]: We should be testing to make sure the link works correctly (i.e. pushes the correct route to history when clicked). There should be plenty of documentation supporting this online so please do it as part of this PR.

Comment (DorianKwan) [reply app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx:50]: +1

File: app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      name: `${header.parent_hash}`,
+    });
+    const validator = screen.getByRole('link', {
+      name: `${body.proposer}`,
+    });
+
+    userEvent.click(parentHash);
+    userEvent.click(validator);
+
+    expect(parentHash).toHaveAttribute('href', `/block/${header.parent_hash}`);
```
Comment (mgibson-casper) [inline app/src/components/cards/BlockDetailsCard/BlockDetailsCard.test.tsx:97]: This is a good first step, but I think we can take it one step further. I believe you should be able to mock out the `history` object (probably with `jest.fn` and then assert that these urls have been pushed there once the link has been clicked.

**Review by mgibson-casper — APPROVED** [review]

**Pull request merged**